File: Data Structures & Algorithms/anagram-groups/submission-2.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        results = {} # frequencies set : list of words
        for word in strs:
            freqs = [0] * 26
            for c in word:
                freqs[ord(c) - ord('a')] += 1
            freqs = tuple(freqs)
            if freqs not in results:
                results[freqs] = [word]
            else:
                results[freqs].append(word)
        res = []
        for words in list(results.values()):
            res.append(words)

        return res

File: Data Structures & Algorithms/is-anagram/submission-2.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        unordered_map<char, int> freqs;
        for (char c : s) {
            if (freqs.find(c) != freqs.end()) {
                freqs[c] += 1;
            } else {
                freqs[c] = 1;
            }
        }

        for (char c : t) {
            if ((freqs.find(c) == freqs.end()) || freqs[c] == 0) {
                return false;
            }
            freqs[c]--;
        }
        return true;
    }
};
